siad/daemon.go

```diff
@@ -109,6 +111,34 @@ func processConfig(config Config) (Config, error) {
 	return config, nil
 }
 
+// walletAutoUnlock is called on siad startup and attempts to automatically
+// unlock the wallet if the SIA_WALLET_PASSWORD env exists on the machine.
+func walletAutoUnlock(w modules.Wallet) {
+	var validKeys []crypto.TwofishKey
+	password := os.Getenv("SIA_WALLET_PASSWORD")
+	if password != "" {
+		fmt.Println("Sia Wallet Password found, attempting to auto-unlock wallet")
+	}
+	dicts := []mnemonics.DictionaryID{"english", "german", "japanese"}
+	for _, dict := range dicts {
+		seed, err := modules.StringToSeed(password, dict)
+		if err != nil {
+			continue
+		}
+		validKeys = append(validKeys, crypto.TwofishKey(crypto.HashObject(seed)))
+	}
+	validKeys = append(validKeys, crypto.TwofishKey(crypto.HashObject(password)))
+	for _, key := range validKeys {
+		err := w.Unlock(key)
+		if err == nil {
+			fmt.Println("Wallet unlocked")
+		}
+		if err != nil && err != modules.ErrBadEncryptionKey {
+			fmt.Println("Automatic unlock failed!")
+		}
+	}
```

Comment:
won't this print `Automatic unlock failed!` for each key in `validKeys` that isn't correct? I think we want:
```go
for _, key := range validKeys {
    if err := w.Unlock(key); err == nil {
        return nil
    }
}
return modules.ErrBadEncryptionKey
```